lib/flags.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+class Flags {
```

Comment:
This nesting looks like too much, maybe there is a possibility of extracting the internal class to possibly decrease indentation?

Reply:
Inside `Flags.from` `FlagsClass` is created which is binded on context of `Flags.from`, to be more specific - on `args`.

Reply:
I'm thinking of the way we did it with the `Enum` class. Create a separate function and assign it to `Flags` later (`Flags.from = createFlags` or `Flags.from = (...args) => ...`).

Reply:
`Enum` has implementation which is similar to mine and I don't like the idea of assigning one method to empty class. If we have to change it like this I'd rather have just single function. Btw, max level of nesting is 4 which is not so bad.